internal/repository/user_login_repository.go

```diff
@@ -32,39 +39,104 @@ func NewUserLoginRepo(db *pgxpool.Pool) UserStorer {
 	}
 }
 
-// GetUserIDByEmail returns and checks the id of specific email
-func (profileStore *UserStore) GetUserIDByEmail(ctx context.Context, email string) (int64, error) {
-
-	var user UserDao
+// GetUserInfo returns and checks the id and role of a specific email or id
+func (userStore *UserStore) GetUserInfo(ctx context.Context, filter specs.UserInfoFilter) (User, error) {
+	var user User
 
-	// used squirrel to build the query
-	selectBuilder := sq.Select("id").From(userTable).Where(sq.Eq{"email": email}).PlaceholderFormat(sq.Dollar)
+	selectBuilder := sq.Select(constants.RequestUserColumns...).From(userTable).PlaceholderFormat(sq.Dollar)
+	if filter.Email != "" {
+		selectBuilder = selectBuilder.Where(sq.Eq{"email": filter.Email})
+	} else if filter.ID > 0 {
+		selectBuilder = selectBuilder.Where(sq.Eq{"id": filter.ID})
+	} else {
+		return User{}, errors.New("filter must contain either email or id")
+	}
 
-	// generate the query
 	selectQuery, args, err := selectBuilder.ToSql()
 	if err != nil {
 		zap.S().Error("Error generating select query: ", err)
-		return 0, err
+		return User{}, err
 	}
 
 	// execute the query using pgx
-	row := profileStore.db.QueryRow(ctx, selectQuery, args...)
-	err = row.Scan(&user.ID)
+	row := userStore.db.QueryRow(ctx, selectQuery, args...)
+	err = row.Scan(&user.ID, &user.Email, &user.Role)
 	if err != nil {
-		if err == sql.ErrNoRows {
-			return 0, errors.ErrNoRecordFound
+		if err == pgx.ErrNoRows {
+			return User{}, errs.ErrNoRecordFound
 		}
 		if helpers.IsDuplicateKeyError(err) {
```

Comment:
I think this check is not required while SELECT query @Abhishek-Jain-1925 

Please remove

Reply:
done